### PR TITLE
Changes for 2.1.1

### PR DESCRIPTION
**Pull request by ThiefMaster**

### CODE REVIEW

File: indico/core/celery/__init__.py [hunk 1]
```diff
@@ -45,7 +63,32 @@ def _import_modules(*args, **kwargs):
...
         return SideMenuItem('celery', _("Tasks"), url_for('celery.index'), 20, icon='time')
+
+
+@template_hook('global-announcement', priority=-100)
+def _inject_announcement_header(**kwargs):
+    if not session.user or not session.user.is_admin or config.DISABLE_CELERY_CHECK:
+        return
+    last_ping = celery_settings.get('last_ping')
+    last_ping_version = celery_settings.get('last_ping_version')
+    down = not last_ping or (now_utc() - last_ping) > timedelta(hours=1)
```
Comment (mic4ael) [inline indico/core/celery/__init__.py]: maybe the interval should be configurable as well?

File: indico/core/celery/__init__.py [hunk 2]
```diff
@@ -45,7 +63,32 @@ def _import_modules(*args, **kwargs):
...
+    mismatch = last_ping_version and last_ping_version != indico.__version__
+    if down or mismatch:
+        return render_template('celery/warning.html', down=down)
+
+
+@celery.periodic_task(name='heartbeat', run_every=crontab(minute='*/30'))
```
Comment (mic4ael) [inline indico/core/celery/__init__.py]: maybe the ping interval should be  configurable as well?

Comment (ThiefMaster) [reply indico/core/celery/__init__.py]: People can already override this via the config, but it's kind of pointless.

File: docs/source/config/settings.rst [hunk 3]
```diff
@@ -388,6 +388,14 @@ Development
 
     Default: ``'https://hub.getindico.io'``
 
+.. data:: DISABLE_CELERY_CHECK
+
+    Disables the warning about Celery not running / being outdated. By default
+    it is disabled when :data:`DEBUG` is enabled, but this setting can be used
+    to explicitly enable/disable the warning regardless of debug mode.
+
+    Default: ``None``
```
Comment (pferreir) [inline docs/source/config/settings.rst]: The default is None?

File: indico/modules/legal/forms.py [hunk 4]
```diff
@@ -26,5 +28,11 @@
 class LegalMessagesForm(IndicoForm):
     network_protected_disclaimer = TextAreaField(_("Network-protected information disclaimer"), widget=CKEditorWidget())
     restricted_disclaimer = TextAreaField(_("Restricted information disclaimer"), widget=CKEditorWidget())
-    tos = TextAreaField(_("Terms and conditions"), widget=CKEditorWidget())
-    privacy_policy = TextAreaField(_("Privacy Policy"), widget=CKEditorWidget())
+    tos_url = URLField(_('URL'), [URL()],
```
Comment (mic4ael) [inline indico/modules/legal/forms.py]: why are you using `'` here and `"` for other strings?

File: docs/source/config/settings.rst (hunk 3 above)
Comment (ThiefMaster) [reply docs/source/config/settings.rst]: See the text above - if unset, it depends on the `DEBUG` option.

File: indico/modules/legal/forms.py (hunk 4 above)
Comment (ThiefMaster) [reply indico/modules/legal/forms.py]: didn't bother changing the existing ones

**Pull request merged**